server/src/test/java/org/elasticsearch/common/collect/ImmutableOpenMapTests.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.collect;
+
+import org.elasticsearch.test.ESTestCase;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class ImmutableOpenMapTests extends ESTestCase {
+
+    ImmutableOpenMap<String, String> regionCurrencySymbols = ImmutableOpenMap.<String, String>builder()
+        .fPut("Japan", "¥")
+        .fPut("USA", "$")
+        .fPut("EU", "€")
+        .fPut("UK", "£")
+        .fPut("Korea", "₩")
+        .build();
+
+    public void testStreamOperationsAreSupported() {
+        assertThat(regionCurrencySymbols.stream().filter(e -> e.key.startsWith("U")).map(e -> e.value).collect(Collectors.toSet()),
+            equalTo(Set.of("£", "$")));
+    }
+
+    public void testSortedStream() {
+        assertThat(regionCurrencySymbols.stream().sorted(Comparator.comparing(e -> e.key)).map(e -> e.value).collect(Collectors.toList()),
+            equalTo(List.of("€", "¥", "₩", "£", "$")));
+    }
+
+    public void testEmptyStreamWorks() {
+        assertThat(ImmutableOpenMap.of().stream().count(), equalTo(0L));
+    }
+}
```

Comment:
Can we add a randomized test too, generating the map input of varying sizes and verifying that streaming though it works and produces the right results. Perhaps also randomly make it sort and/or parallel.

Reply:
Will add a couple of random tests and check if something gets broken!